interleave-in-parent.md

```diff
@@ -0,0 +1,172 @@
+
+### Key-Value Structure
+
+When you write data into the child table, it is inserted into the key-value store immediately after the parent table's key matching the interleave prefix.
+
+For example, if you interleave `orders` into `customers`, the `orders` data is written directly within the `customers` table in the key-value store. The following is a crude, illustrative example of what the keys would look like in this structure:
+
+~~~
+/customers/1
+/customers/1/orders/1000
+/customers/1/orders/1002
+/customers/2
+/customers/2/orders/1001
+/customers/2/orders/1003
+...
+/customers/n/
+/customers/n/orders/<order belonging to n>
+~~~
+
+By writing data in this way, related data is more likely to remain on the same key-value range, which can make it much faster to read from and write to. Using the above example, all of customer 1's data is going to be written to the same range, including its representation in both the `customers` and `orders` tables.
+
+## When should I interleave tables?
+
+You're most likely to benefit from interleaved tables when:
+
+  - Your tables form a [hierarchy](#interleaved-hierarchy)
+  - Queries maximize the [benefits of interleaving](#benefits)
+  - Queries do not suffer too greatly from interleaving's [tradeoffs](#tradeoffs)
+
+### Interleaved Hierarchy
+
+Interleaved tables typically work best when the tables form a hierarchy. For example, you could interleave the table `orders` (as the child) into the table `customers` (as the parent, which represents the people who placed the orders). You can extend this example by also interleaving the tables `invoices` (as a child) and `packages` (as a child) into `orders` (as the parent).
+
+The entire set of these relationships is referred to as the **interleaved hierarchy**, which are related through the interleave prefix columns.
+
+### Benefits
+
+In the general case, reads, writes, and joins of data related through the interleave prefix are *much* faster. However, you can also improve performance with any of the following:
+
+- Filtering more columns in the interleave prefix (from left to right).<br/><br/>For example, if the interleave prefix of `packages` is `(customer, order)`, filtering on `customer` would be fast, but filtering on `customer` *and* `order` would be faster.
+- Using only tables in the interleaved hierarchy.
+
+### Tradeoffs
+
+- In the general case, reads and deletes over ranges of data (e.g., `WHERE col > val`) in interleaved tables are slower.<br/><br/>However, an exception to this is performing range operations on the greatest descendent in the interleaved hierarchy that filters on all columns of the interleave prefix with constant values.<br/><br/>For example, if the interleave prefix of `packages` is `(customer, order)`, filtering on the entire interleave prefix with constant values while using range operations on another column, like `WHERE customer = 1 AND order = 1001 AND delivery_date > DATE '2016-01-25'`, would still be fast.
```

Comment:
nit: "In the general case" > "In general".

I don't understand the `WHERE col > val` example. How is that supposed to represent "reads and deletes over ranges of data"? Also, what do we mean by "ranges of data"? Is it possible to phrase this in SQL-specific terms rather than key-value terms, if that's what we're in fact doing?

Same concern about term "range operations" in the rest of this paragraph.